nucypher/device/base.py

```diff
@@ -0,0 +1,96 @@
+"""
+This file is part of nucypher.
+
+nucypher is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+nucypher is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with nucypher.  If not, see <https://www.gnu.org/licenses/>.
+"""
+
+
+from abc import ABC, abstractmethod
+from collections import namedtuple
+
+
+class TrustedDevice(ABC):
+    """
+    An Abstract Base Class for implementing wallet-like functions for stakers
+    utilizing trusted hardware devices (e.g. trezor).
+    This class specifies a few basic functions required for staking on the
+    NuCypher network.
+
+    TODO: Define an abstractmethod for BIP44 derivation?
+    """
+
+    ETH_BIP44_PATH = "m/44'/60'/0'/0/{address_index}"
```

Comment:
Do we want to use a `chainID` here for separation of testnet keys (the first `0` can be a `5` for example)?

Reply:
I was thinking about this, but I'm unsure if I want to support it quite yet. Ideally, you configure a device _for a testing chain_ and _do not_ allow the user to use the _same_ device on _different_ chains.

Reply:
In this case - why default to chain ID `0`?

Reply:
Because there is no interface for a device config yet and it doesn't really matter which chain is derived -- it works on both.

Reply:
We can certainly hardcode to Goerli, if that makes more sense for now. I'm not sure if web wallets actually follow this, though...